dataplane/api-server/src/server.rs

```diff
@@ -51,6 +54,40 @@ impl BackendService {
         backends_map.remove(&key)?;
         let mut gateway_indexes_map = self.gateway_indexes_map.lock().await;
         gateway_indexes_map.remove(&key)?;
+
+        // Delete all entries in our tcp connection tracking map that this backend
```

Comment:
Do we need to think about adding a "timeout" at some point so that stale threads are automatically pruned?

Reply:
Correct, I'm not even sure we need that mechanism yet just thinking about it, Let's defer for a follow up.